core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -113,12 +146,36 @@ var TopicsAndSkillsDashboardPage = function() {
     assignSkillToTopicButtons.then(function(elems) {
       elems[index].click();
       topicsListItems.then(function(topics) {
-        topics[index].click();
+        topics[topicIndex].click();
         confirmMoveButton.click();
       });
     });
   };
 
+  this.assignSkillWithDescriptionToTopicByTopicName = async function(
+      skillDescription, topicName) {
+    var elems = await getAssignSkillToTopicButtons();
+    var skills = await getSkillDescriptions();
+    for (var i = 0 ;i < skills.length;i++) {
+      var skillText = await skills[i].getText();
+      if (skillText === skillDescription) {
+        await elems[i].click();
+        var topics = await getTopicsListItems();
+        var topicNames = await getTopicNamesInTopicSelectModal();
+        for (var j = 0; j < topicNames.length; j++) {
+          var topicText = await topicNames[j].getText();
+          if (topicText === topicName) {
+            await topics[j].click();
+            await confirmMoveButton.click();
+          }
+        }
+      }
+    }
+    return new Promise(function(resolve, reject) {
+      resolve('done');
+    });
```

Comment:
Although the expected behavior of the main test function(in `topicAndStoryEditor.js` ) is to wait for the function using async-await(in  `protactor_utils.`).
The problem is that the control flow of the main test function in `topicAndStoryEditor.js` doesn't wait for the functions using async-await in `protactor_utils.`(as we discussed async-await and control flow don't go well together.) to get completed.
But async-await works well inside the individual functions in `protactor_utils.`. 
So, I made the main `it(') ` block wait until the async-await is finished and the promise is resolved.
By adding the async-await inside a promise and making the main function wait(using `browser.wait`). , we make sure that the control flow doesn't proceed to the next function without the function using async-await is completed.
The only other option is to migrate everything to async-await(since the PR #9267 is opened).

@U8NWXD any suggestions?

Reply:
@donosco98 Ahh I see. This seems like a reasonable approach then. If this PR goes in before we finish the migration then the migration can clean this up. Otherwise, we can change this after the migration is done.